drivers/volume/portworx/portworx.go

```diff
@@ -280,18 +292,21 @@ func (d *portworx) isMetadataNode(node node.Node, address string) (bool, error)
 }
 
 func (d *portworx) CleanupVolume(name string) error {
-	locator := &api.VolumeLocator{}
-
-	volumes, err := d.getVolDriver("").Enumerate(locator, nil)
+	volumes, err := d.getVolDriver().Enumerate(d.getContext(""), &api.SdkVolumeEnumerateRequest{}, nil)
 	if err != nil {
 		return err
 	}
 
-	for _, v := range volumes {
+	for _, volID := range volumes.GetVolumeIds() {
+		volumeInspectResponse, err := d.getVolDriver().Inspect(d.getContext(""), &api.SdkVolumeInspectRequest{VolumeId: volID})
```

Comment:
we have different situations here, sometimes when it comes from k8s we have the volumeName, when it comes from px we have volumeId, in both cases the sdk call will use it as VolumeId in request but inside pwx it will look for both fields volumeId and volume name, we cannot change some case since we cannot use pwx id to find a volume in k8s but we can use volume name to find a volume in pwx
i.e
in pwx
```
ID			NAME						SIZE	HA	SHARED	ENCRYPTED	IO_PRIORITY	STATUS				SNAP-ENABLED
1047224954423951661	pvc-66e93073-21c8-11ea-9796-000c2962ba54	5 GiB	3	no	no		HIGH		up - attached on 70.0.87.44	no
```
in  k8s
```
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                              STORAGECLASS   REASON   AGE
pvc-66e93073-21c8-11ea-9796-000c2962ba54   5Gi        RWO            Delete           Bound    default/rabbitmq-data-rabbitmq-0   rabbitmq-sc             17h
```